File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int ptr = 0;

        for(int i = 0; i< nums.size(); i++) {
            if (i == 0 || nums[i] != nums[i-1]) {
                nums[ptr++] = nums[i];
            }
        }
        nums.resize(ptr);
        return ptr;

    }
};

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-1.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        l = 0
        currSum = 0
        ans = len(nums) + 1
        for r, num in enumerate(nums):
            currSum += num
            if currSum >= target:
                while currSum - nums[l] >= target:
                    currSum -= nums[l]
                    l += 1
                ans = min(ans, r - l + 1)
        return ans if ans != len(nums) + 1 else 0
